### DISCUSSION TITLE
How to handle Automated 2FA

### DISCUSSION THREAD

**Discussion by odhiambo** [mrlt8/docker-wyze-bridge#570] [General]
Hello friends.

The installation was quite easy in my HA instance which runs in docker. Configuration is the one giving me a problem :-)

I am using the Time-based One-Time Password (TOTP), and I understand that I can have the bridge generate and enter the verification when needed by setting TOTP_KEY to my shared secret/secret key - the one that was provided in the Wyze app when setting up TOTP.
I am assuming that should go into  /config/wyze-bridge/mfa_token.txt., but I can't figure out how.
Should I enter in one line, colon-separated or space-separated?
TOTP_KEY:  secret key ?
TOTP_KEY secret key
TOTP_KEY=secret key

Thank you in advance.

**Comment by odhiambo** [comment]
Just to add, here is the log from the add-on:

2022/10/22 19:19:01 [WyzeBridge] 🚀 STARTING DOCKER-WYZE-BRIDGE v1.8.10
2022/10/22 19:19:01 [WyzeBridge] 🏠 Home Assistant Mode
2022/10/22 19:19:02 [WyzeBridge] 🔍 Could not find local cache for 'user'
 * Serving Flask app 'frontend'
2022/10/22 19:19:02 [WyzeBridge] 🔍 Could not find local cache for 'auth'
2022/10/22 19:19:02 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
 * Debug mode: off
2022/10/22 19:19:02 [WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://172.30.33.3:5000
2022/10/22 19:19:02 [WyzeBridge] Press CTRL+C to quit
2022/10/22 19:19:03 [WyzeBridge] 🚷 Invalid credentials?
2022/10/22 19:19:03 [WyzeBridge] 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/user/login

**Comment by mrlt8** [comment]
You should be able to set the key in `TOTP_KEY` in the configuration tab for the bridge. You may need to toggle "Show unused optional configuration options".
or if you'd like to set it locally, you can create a file `/config/wyze-bridge/totp` and save the key there.

**Reply by odhiambo** [reply]
Thank you very much for the clarification. 
I had to disable the @2FA and re-enable it to get the secret key from the Wyze app. Once I put that in and restarted the addon, all my cameras were shown in the log. But there is that warning about "WSGI server" that I do not understand. 
Is that about an alternative deployment method?

2022/10/23 13:12:43 [WyzeBridge] 🚀 STARTING DOCKER-WYZE-BRIDGE v1.8.11
2022/10/23 13:12:43 [WyzeBridge] 🏠 Home Assistant Mode
2022/10/23 13:12:43 [WyzeBridge] 📚 Using 'user' from local cache...
 * Serving Flask app 'frontend'
 * Debug mode: off
_**2022/10/23 13:12:43 [WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.**_
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://172.30.33.3:5000
2022/10/23 13:12:43 [WyzeBridge] Press CTRL+C to quit
2022/10/23 13:12:43 [WyzeBridge] 📚 Using 'cameras' from local cache...
2022/10/23 13:12:43 [WyzeBridge] 
🎬 STARTING ALL 4 CAMERAS
2022/10/23 13:12:44 [WyzeBridge] Starting rtsp-simple-server v0.20.0
2022/10/23 13:12:44 [WyzeBridge] [ON-DEMAND] ⌛️ WyzeCam V3 - front-porch-cam on 10.10.12.216 (1/3)
2022/10/23 13:12:44 [WyzeBridge] [ON-DEMAND] ⌛️ WyzeCam Doorbell - front-door on 10.10.12.189 (1/3)
2022/10/23 13:12:44 [WyzeBridge] [ON-DEMAND] ⌛️ WyzeCam Pan V2 - front-door-cam on 10.10.12.213 (1/3)
2022/10/23 13:12:44 [WyzeBridge] [ON-DEMAND] ⌛️ WyzeCam Pan - living-room-cam on 10.10.12.159 (1/3)

**Reply by mrlt8** [reply]
That's coming from flask, but you can safely ignore it.

**Reply by MomosX** [reply]
> You should be able to set the key in `TOTP_KEY` in the configuration tab for the bridge. You may need to toggle "Show unused optional configuration options". or if you'd like to set it locally, you can create a file `/config/wyze-bridge/totp` and save the key there.

Hi,
I'm on 2.10.2 version of the addon.  I do not see any 'TOTP_KEY' in the config tab.   Show unused is checked. Does it have a different name please ? 


Whatever i do the addon does not connect , coming back with 
"🚀 DOCKER-WYZE-BRIDGE v2.10.2 X86_64

[WyzeBridge] 🔍 Could not find local cache for 'auth'
[WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
 * Serving Flask app 'frontend'
 * Debug mode: off
[WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://172.30.33.13:5000
[WyzeBridge] Press CTRL+C to quit
[WyzeBridge] [API] code='1000' msg='Invalid credentials, please check username, password, keyid or apikey' method=POST path=/api/user/login
[WyzeBridge] [API] Clearing credentials. Please try again.
[WyzeBridge] [API] Cool down for 20s before trying again.
[WyzeBridge] Credentials required to complete login!
[WyzeBridge] Please visit the WebUI to enter your credentials."

This is addon conf:
<img width="464" alt="image" src="https://github.com/user-attachments/assets/d7a74902-8987-430b-b173-2a3c3e5b9aa3">

The Api Key and ID are correct.  I even generated a new one.
One thing i should mention is that my Wyze account password contains a "@".  I tried doubling the character, still no luck.

Any idea please on how to get it working ?

Many thanks !

**Reply by mrlt8** [reply]
@MomosX `TOTP_KEY` has been deprecated as it is no longer supported by the wyze authentication API.

Can you try visiting the webUI of the bridge to see if you can complete the wyze authentication there?

**Reply by MomosX** [reply]
Hi ! Thank you for the swift reply :) 

what i did meanwhile:  changed my Wyze account password to get rid of any special character.  Created new API Key.
When starting the addon i still get the exact message.  
But going to the webUI of the bridge works now :)  
Does not survive any reboot or addon restart.

**Reply by MomosX** [reply]
@mrlt8 I figured it out !! all working as expected now.  Just need to learn to properly use this wonderfull addon :) 

Thank you !
